sdk/core/Azure.Core.Experimental/src/CloudMachine.cs

```diff
@@ -0,0 +1,253 @@
+    public class CloudMachine
+    {
+        /// <summary>
+        /// Unique identifier of a CloudMachine. It's the name of the resource group of all the CloudMachine resources.
+        /// </summary>
+        public string Id { get; private set; }
+
+        /// <summary>
+        /// Friendly name of CloudMachine. It's stored as a Tag of all Azure resources associated with the machine.
+        /// </summary>
+        public string DisplayName { get; set; }
+
+        /// <summary>
+        /// Azure subscription ID.
+        /// </summary>
+        public string SubscriptionId { get; private set; }
+
+        /// <summary>
+        /// Azure region, e.g. westus2
+        /// </summary>
+        public string Region { get; private set; }
+
+        private int Version { get; set; }
+
+        /// <summary>
+        /// Creates a new CloudMachine
+        /// </summary>
+        /// <param name="subscriptionId"></param>
+        /// <param name="region"></param>
+        /// <returns></returns>
+        /// <remarks>DisplayName is initialized to id. It can be changed by setting the DisplayName property.</remarks>
+        public static CloudMachine Create(string subscriptionId, string region)
+        {
+            if (string.IsNullOrEmpty(subscriptionId)) throw new ArgumentNullException(nameof(subscriptionId));
+            if (string.IsNullOrEmpty(region)) throw new ArgumentNullException(nameof(region));
+
+            var id = GenerateCloudMachineId();
+            var defaultDisplayName = $"{id}@{DateTime.UtcNow:d}";
+            var cm = new CloudMachine(id, defaultDisplayName, subscriptionId, region, 1);
+            return cm;
+        }
+
+        /// <summary>
+        /// Loads CloudMachine settings from configurationFile
+        /// </summary>
+        /// <param name="configurationFile"></param>
+        /// <exception cref="InvalidCloudMachineConfigurationException"></exception>
+        public CloudMachine(string configurationFile = ".\\cloudconfig.json")
+        {
+            try
+            {
+                byte[] configurationContent = File.ReadAllBytes(configurationFile);
+                var document = JsonDocument.Parse(configurationContent);
+                JsonElement json = document.RootElement.GetProperty("CloudMachine");
+
+                Id = ReaderString(json, "id", configurationFile);
+                SubscriptionId = ReaderString(json, "subscriptionId", configurationFile);
+                Region = ReaderString(json, "region", configurationFile);
+                DisplayName = ReaderString(json, "name", configurationFile);
+                Version = ReaderInt32(json, "version", configurationFile);
+            }
+            catch (InvalidCloudMachineConfigurationException)
+            {
+                throw;
+            }
+            catch (Exception e)
+            {
+                throw new InvalidCloudMachineConfigurationException(configurationFile, setting: null, e);
+            }
```

Comment:
nit: can simplify to
```suggestion
            catch (Exception e) when (e is not InvalidCloudMachineConfigurationException)
            {
                throw new InvalidCloudMachineConfigurationException(configurationFile, setting: null, e);
            }
```